dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+			return null;
+		}
+		String idInode = (String) request.getAttribute("idInode");
+		Identifier id;
+
+		try {
+			id=APILocator.getIdentifierAPI().find(idInode);
+		} catch (DotDataException e1) {
+			Logger.warn(VelocityUtil.class, "can't load page identifier",e1);
+			return null;
+		}
+
+		IHTMLPage page = getPage(id, request, false, null);
+		if (page == null || page.getCacheTTL() < 1) {
+			return null;
+		}
+		StringBuilder sb = new StringBuilder();
+		sb.append(page.getInode());
+		sb.append("_" + page.getModDate().getTime());
+		return sb.toString();
+	}
+
+	public static long getLanguageId(HttpServletRequest request) {
+		return WebAPILocator.getLanguageWebAPI().getLanguage(request).getId();
+	}
+
+	/**
+	 * Retrieves the list of languages a given Content Page ({@link Contentlet})
+	 * is available on. This is useful for keeping users from selecting a page
+	 * language that has no associated content at the moment.
+	 *
+	 * @param contentlet
+	 *            - The Content Page.
+	 * @return The {@link List} of languages the content page is available on.
+	 */
+	public static List<DisplayedLanguage> getAvailableContentPageLanguages(
+			Contentlet contentlet) {
+		List<DisplayedLanguage> languages = new ArrayList<DisplayedLanguage>();
+		List<DisplayedLanguage> allDisplayLanguages = new ArrayList<DisplayedLanguage>();
+
+		boolean doesContentHaveDefaultLang = false;
+
+		for (Language language : APILocator.getLanguageAPI().getLanguages()) {
+			if (language.getId() != contentlet.getLanguageId()) {
+				try {
+					APILocator.getContentletAPI()
+							.findContentletByIdentifier(
+									contentlet.getIdentifier(), false,
+									language.getId(),
+									APILocator.getUserAPI().getSystemUser(),
+									false);
+				} catch (Exception e) {
+					Logger.debug(
+					        VelocityUtil.class,
+							"The page is not available in language "
+									+ language.getId() + ". Just keep going.");
+
+					if(DEFAULT_PAGE_TO_DEFAULT_LANGUAGE) {
+						allDisplayLanguages.add(new DisplayedLanguage(language, true));
+					}
+
+					continue;
+				}
+			}
+			if(language.getId()==APILocator.getLanguageAPI().getDefaultLanguage().getId()) {
+				doesContentHaveDefaultLang = true;
+			}
+
+			languages.add(new DisplayedLanguage(language, false));
+
+			if(DEFAULT_PAGE_TO_DEFAULT_LANGUAGE) {
+				allDisplayLanguages.add(new DisplayedLanguage(language, false));
+			}
+		}
+
+		if(DEFAULT_PAGE_TO_DEFAULT_LANGUAGE && doesContentHaveDefaultLang){
+			return allDisplayLanguages;
+		}
+
+		return languages;
+	}
+
+	/**
+	 * This returns the proper ihtml page based on id, state and language
+	 * @param id
+	 * @param request
+	 * @param live
+	 * @return
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
+	public static IHTMLPage getPage(Identifier id, HttpServletRequest request, boolean live, Context context) throws DotDataException, DotSecurityException{
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 153 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)